src/map.h

```diff
@@ -20,6 +20,7 @@
 #include "item_stack.h"
 #include "active_item_cache.h"
 #include "string_id.h"
+#include "veh_type.h"
```

Comment:
Try to avoid including headers (if at all possible), especially in a header. This only creates dependencies and increase compile time. In this case: if somebody changes veh_type.h, that would also imply a change to map.h (as it includes veh_type.h) and all files that include map.h (quite a lot) need to be recompiled.

It seems you only need the definition of `vgroup_id`, you can forward declare it like this:
```C++
class vehicle_group;
using vgroup_id = string_id<vehicle_group>;
```